### DISCUSSION TITLE
User login not working for old database

### DISCUSSION THREAD

**Discussion by galvakojis** [fastapi-users/fastapi-users#994] [Q&A]
So I have migrated form version 5 to 10, but login works only if I am using new collection `User` if I switch to old one `users` route `/api/users/me` returns Unauthorized even after I log in. Not sure if this is a bug, or am I missing something?

**Comment by frankie567** [comment]
From what you wrote, I understand that you were using MongoDB, right?

On V10, Users will use the ID generated by MongoDB, an ObjectID. Before, we were using UUID. It's probable that your old collection has the `id` property on every user which is still an UUID.

Two solutions for you:

1. Stay with UUID. In this case, you need to overload the `id` property on the Beanie model:

```py
import uuid

from pydantic import Field

class User(BeanieBaseUser[uuid.UUID]):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
```

You'll also need to use a UUID parser on `UserManager` instead of `ObjectIDIDMixin`:

```py
from fastapi_users import BaseUserManager, UUIDIDMixin

class UserManager(UUIDIDMixin, BaseUserManager[User, uuid.UUID]):
    reset_password_token_secret = SECRET
    verification_token_secret = SECRET
    # ...
```

2. Use native `ObjectID`. In this case, I think the solution is to remove the `id` field on your collection to just keep the native `_id`. For example: 

```js
db.users.updateMany({}, {$unset: {id: ""}})
```

**MAKE SURE YOU HAVE A BACKUP OF YOUR DATA IN CASE THINGS GO WRONG**

However, if you have other collections which used the UUID to reference the user, you'll have to update them so they reference them by `_id`.

**Reply by galvakojis** [reply]
Yes, you were right
was using this ` class UserManager(**ObjectIDIDMixin**, BaseUserManager[User, uuid.UUID]):`

insted of` class UserManager(**UUIDIDMixin**, BaseUserManager[User, uuid.UUID]):`

**Comment by galvakojis** [comment]


**Reply by frankie567** [reply]
Check that your `UserManager` has the right `UUIDIDMixin`:

```py
class UserManager(UUIDIDMixin, BaseUserManager[User, uuid.UUID]):
    reset_password_token_secret = SECRET
    verification_token_secret = SECRET
    # ...
```

PS: Your Beanie model should also be like this (not the cause of the issue, but better for typing):

```py
class User(BeanieBaseUser[uuid.UUID]):
    # ...
```
